### DISCUSSION TITLE
Add default service providers in configuration

### DISCUSSION THREAD

**Discussion by Yoji** [joomla/joomla-cms#39209] [General]
![изображение](https://user-images.githubusercontent.com/6072785/173077827-5a3a21fc-20ef-438b-af8d-e6f6f28b02a7.png)

It would be nice to add the ability to replace and extend standard service providers in the configuration file.

For example, I need to extend the standard UserFactory, but I can't replace it before starting the Application and creating the container.

**Comment by laoneo** [comment]
You should be able to write a system event and then listen to the `onBeforeExecute` event and then set it on the container.

**Comment by Yoji** [comment]
> You should be able to write a system event and then listen to the `onBeforeExecute` event and then set it on the container.

`onBeforeExecute` is a potential problem.
For example, several plugins are installed on the site that use `onBeforeExecute' and one of them uses the User provider before it was replaced

**Comment by brianteeman** [comment]
Plugins are executed in order

**Comment by joomdonation** [comment]
I don't see any chance this request will be added to the core, so please use a system plugin to handle it as suggested by @laoneo above. I will convert this to discussion so that we can discuss further from there if needed, maybe it could be good ideas for future releases of Joomla.
